### ISSUE TITLE
📚 Documentation: Deep dive authentication recipe with Vue

### ISSUE THREAD

**Issue by gewenyu99** [appwrite/website#90]
### Goal
Deep dive tutorial for beginners to learn Appwrite Authentication in depth with Vue.

### Content
1. The end product should be a functional app with auth screens.
2. Step by step from setup. If there will be test data, provide a Server SDK script with instructions.
3. Format should be similar to ` /docs/tutorials/sveltekit`
4. Should contain the following content similar to [this PR](https://github.com/appwrite/website/pull/137)
      1. Sign up
      1. Log in
      1. Verification
      1. Recovery
      1. Conditional routing based on auth state
      1. Preferences
      1. Delete account

### How to contribute
Pitch an outline detailing each page you plan to add and headings within, and ask to be assigned. Open a PR **after** the outline has been approved.

The Appwrite team will collaborate with you on the outline and PR.

Thanks for taking the time to make our documentation better 🙏

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by ayan-joshi** [comment]
I want to work on this issue  , Can you please assign this to me @Haimantika @gewenyu99

**Comment by Haimantika** [comment]
> I want to work on this issue , Can you please assign this to me @Haimantika @gewenyu99

Hi @ayan-joshi have assigned the issue to you! Thank you for showing interest in contributing to Appwrite! Happy Hacktoberfest 🎃

Notes:

Please update us with your progress every 3 days, so that we know that you are working on it.
Join us on Discord - https://appwrite.io/discord to chat about Hacktoberfest and Appwrite!

**Comment by Haimantika** [comment]
> I want to work on this issue , Can you please assign this to me @Haimantika @gewenyu99

Hi @ayan-joshi I am unassigning you from this issue, as I have already assigned you two issues. This will leave room for more people to participate, thanks for you interest

**Comment by Vaibhav91one** [comment]
I want to work on this issue @Haimantika.

**Comment by Haimantika** [comment]
> I want to work on this issue @Haimantika.

Hi, assigned the issue to Ayan now, since we follow first-come first-serve basis, if he doesn’t follow up in 3 days, will assign it to you.

**Comment by ayan-joshi** [comment]
> > I want to work on this issue @Haimantika.
> 
> Hi, assigned the issue to Ayan now, since we follow first-come first-serve basis, if he doesn’t follow up in 3 days, will assign it to you.

Thanks

**Comment by gewenyu99** [comment]
@ayan-joshi You'll be following a similar structure and content to this PR: https://github.com/appwrite/website/pull/137

Before starting, please provider an outline of:
- Every page in your contribution
- Every heading in your page
- Why you need these headings and what will be in them.
Thanks!

**Comment by Haimantika** [comment]
Hi @ayan-joshi is there an update?

**Comment by gewenyu99** [comment]
@Vaibhav91one It's yours now :)

**Comment by gewenyu99** [comment]
@ayan-joshi No problem! Thanks for your help with the Ionic app!

**Comment by Vaibhav91one** [comment]
> @ayan-joshi No problem! Thanks for your help with the Ionic app!

Will be sharing a outline of the application soon.

**Comment by Vaibhav91one** [comment]
@gewenyu99 I have come up with this outline. Please provide feedback and provide the green light.

# Page Outline

## Common Components

### Header:
- Logo
- Navigation bar
  
### Footer:
  - Copyright information
  - Links to social media pages
  - Register Page

## Login Page:
- Login Form
- Button to submit
- Link for register Page

## Register Page:
- Register Form
- Button to submit registration
- Link for login Page

## Dashboard
- Will Provide all the details that were filled
- Some additional things etc

## Will be focusing on the above for now and  later will add Verification, Recovery, Delete account functionality.

**Comment by gewenyu99** [comment]
Hi, I decided to do an example project. Please follow the outline and format here:
https://github.com/appwrite/website/issues/89

**Comment by mecskyverse** [comment]
Hey, @Vaibhav91one, are you currently working on this issue? It's been a long time since your last follow-up.

**Comment by Vaibhav91one** [comment]
Sorry, about not providing with an update. Will be providing an update tomorrows (20th oct). Have been a little busy with college.

**Comment by mecskyverse** [comment]
Hey, @gewenyu99 @Haimantika can I get to work on this issue.

**Comment by gewenyu99** [comment]
@Vaibhav91one Can I get a quick status update? :D

**Comment by Vaibhav91one** [comment]
Yup will be making a PR today and If I don't, you can assign it to @mecskyverse

**Comment by Vaibhav91one** [comment]
Okay @gewenyu99 I am on step 5. I have covered about creating the app, setting up store, routing, login, logout, register, Some routing restriction. Now, I will be focusing on verification, updating, forgot password.

The app is looking like this if  the tutorial is followed. (not the end product)
![image](https://github.com/appwrite/website/assets/108279746/7091a0a8-7fa1-4aa9-9818-360e0625df81)
![image](https://github.com/appwrite/website/assets/108279746/8783d970-1320-40db-b8a7-f76ab6d79ef3)
![image](https://github.com/appwrite/website/assets/108279746/80da2c26-0a34-4384-ba1a-1573f8555dfe)
![image](https://github.com/appwrite/website/assets/108279746/c869be09-a79b-4367-9d8e-e5c9e8c9b3df)
![image](https://github.com/appwrite/website/assets/108279746/20d3b213-8f52-4d88-af83-f0c15639dcb0)

A feedback will be much appreciated.

**Comment by Vaibhav91one** [comment]
Update, So, I have implemented verification, password recovery and navigation guards to the app. Will be writing it down in the tutorial. 

I am having a little trouble with account delete functionality. @gewenyu99. little help will be much appreciated.

**Comment by gewenyu99** [comment]
> Update, So, I have implemented verification, password recovery and navigation guards to the app. Will be writing it down in the tutorial.
> 
> I am having a little trouble with account delete functionality. @gewenyu99. little help will be much appreciated.

So you can only block your account on the client side. 

I would do a block and add a comment that this is to protect user data, once they block their account, there should be server side scripts ran periodically to delete associated data that contains the userID.

**Comment by gewenyu99** [comment]
> Okay @gewenyu99 I am on step 5. I have covered about creating the app, setting up store, routing, login, logout, register, Some routing restriction. Now, I will be focusing on verification, updating, forgot password.
> 
> The app is looking like this if the tutorial is followed. (not the end product) ![image](https://user-images.githubusercontent.com/108279746/277766376-7091a0a8-7fa1-4aa9-9818-360e0625df81.png) ![image](https://user-images.githubusercontent.com/108279746/277766555-8783d970-1320-40db-b8a7-f76ab6d79ef3.png) ![image](https://user-images.githubusercontent.com/108279746/277766658-80da2c26-0a34-4384-ba1a-1573f8555dfe.png) ![image](https://user-images.githubusercontent.com/108279746/277766826-c869be09-a79b-4367-9d8e-e5c9e8c9b3df.png) ![image](https://user-images.githubusercontent.com/108279746/277766956-20d3b213-8f52-4d88-af83-f0c15639dcb0.png)
> 
> A feedback will be much appreciated.

Looks great, but please don't include style/css in the tutorials. It's out of scope and distracting. If you must, you can use [Pink Design](https://pink.appwrite.io/) but we prefer if you used none at all :)

**Comment by Vaibhav91one** [comment]
@gewenyu99 Okay will add do the addition and changes. First I included some styles, later I though it is out of scope only, So, I removed it.

**Comment by Vaibhav91one** [comment]
@gewenyu99  I have made the PR. Please provide feedback (if any) 🤞. 
https://github.com/appwrite/website/pull/301

Also, I do have created one PR  a few weeks ago. Also, please review it. 
https://github.com/appwrite/sdk-for-cli/pull/99

**Comment by ChiragAgg5k** [comment]
Closing — the Vue authentication deep-dive recipe was added in https://github.com/appwrite/website/pull/301.
